python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1036,7 +1047,16 @@ def _finishIngest(
                 record_validation_info=record_validation_info,
             )
             refsAndInfos.extend([(ref, info) for ref in dataset.refs])
-        self._register_datasets(refsAndInfos)
+            for ref in dataset.refs:
+                if ref.id.version != 5:
+                    uses_uuid_v5 = False
```

Comment:
What happens if there is a mix of v4/v5 UUIDs? Should we split them into two groups? If we don't care then maybe add `break` after this assignment?

Reply:
Hmm. Maybe I should split them into groups. I'm not expecting it to happen but you are right that it's the consistent thing to do with the current brokenness.

Reply:
I now split them up.